stdlib/Test/src/Test.jl

```diff
@@ -200,7 +200,7 @@ function Base.show(io::IO, t::Fail)
         if data !== nothing
             # The test was an expression, so display the term-by-term
             # evaluated version as well
-            print(io, "\n   Evaluated: ", data)
+            print(io, "\n   Evaluated: ", repr(data))
```

Comment:
I don't know the full context of the error here, but all of the fields (including `data`) are already a String, so it won't change the correct output here. You may want to instead change the `string` call in the constructor to call `repr` there instead. But that may not work either, since some many `eval_test` outputs already do some processing to form the evaluated result expecting it to be passed to `string`. So I think `eval_test` is actually where this handling is missing and where you need to handle this: wrap each argument there in a call to `Base.quoted` inside that function.